index.js

```diff
@@ -48,7 +48,12 @@ CsvWriteStream.prototype._compile = function(headers) {
 CsvWriteStream.prototype._transform = function(row, enc, cb) {
   var isArray = Array.isArray(row)
 
-  if (!isArray && !this.headers) this.headers = Object.keys(row)
+  if (!this.headers) {
+    this.headers = Object.keys(row)
```

Comment:
this is kinda weird to set `this.headers` even when it's an array... maybe we should do this logic down at line 77?